src/main/kotlin/com/hibob/academy/employee_feedback_feature/dao/FeedbackDao.kt

```diff
@@ -23,6 +24,41 @@ class FeedbackDao @Autowired constructor(private val sql: DSLContext) {
         )
     }
 
+    fun getFeedbacksUsingFilter(filter: FeedbackFilter): List<FeedbackData> {
+        var allFeedbacksByCompanyId = selectFeedbacksByCompanyId(filter.companyId)
+
+        filter.isAnonymous?.let {
+            allFeedbacksByCompanyId = allFeedbacksByCompanyId.and(feedback.isAnonymous.eq(it))
+        }
+
+        filter.department?.let {
+            allFeedbacksByCompanyId = allFeedbacksByCompanyId.and(feedback.department.eq(it))
+        }
+
+        filter.createdAt?.let {
+            allFeedbacksByCompanyId = allFeedbacksByCompanyId.and(feedback.createdAt.between(it, LocalDateTime.now()))
```

Comment:
Just making sure - this filter from date X to now, is this what you want the code to do?